go/libraries/doltcore/sqle/database.go

```diff
@@ -321,6 +321,14 @@ func (db Database) getTableInsensitive(ctx *sql.Context, head *doltdb.Commit, ds
 	//  at runtime
 	switch {
 	case strings.HasPrefix(lwrName, doltdb.DoltDiffTablePrefix):
+		if resolve.UseSearchPath && db.schemaName == "" {
+			schemaName, err := resolve.FirstExistingSchemaOnSearchPath(ctx, root)
```

Comment:
This isn't right, should be resolving the table name embedded in the `dolt_diff` table name via `resolve.Table`. Same for the rest of these. Also add some tests to capture this behavior.